monix-eval/shared/src/main/scala/monix/eval/internal/TaskBracket.scala

```diff
@@ -190,7 +190,7 @@ private[monix] object TaskBracket {
                 val fb =
                   try use(value)
                   catch { case NonFatal(e) => Task.raiseError(e) }
-                fb.flatMap(releaseFrame)
+                fb.flatMap(releaseFrame).flatMap(r => Task { conn.pop(); r })
```

Comment:
I feel like we could call `ctx.connection.pop()` in `unsafeRecover` and `unsafeApply` which I removed by mistake in the PR you mentioned

Reply:
If I put them there tests fail. I think it breaks the feature you did in that PR.

Reply:
Ah, I see.

I think we could do it in a similar way to Cats IO: https://github.com/typelevel/cats-effect/blob/master/core/shared/src/main/scala/cats/effect/internals/IOBracket.scala#L189

so

```scala
  private[this] val disableUncancelableAndPop: (Any, Throwable, Context, Context) => Context =
    (_, _, old, _) => {
      old.connection.pop()
      old
    }
```

instead of a `null` in `ContextSwitch`.

I'm not able to check right now but the testa should pass since they are passing for Cats IO and hopefully the memory leak will go away :)

Reply:
Sure, it works fine. Thank you!